compiler/optimizer/OMRRecognizedCallTransformer.cpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+* Copyright (c) 2017, 2017 IBM Corp. and others
+*
+* This program and the accompanying materials are made available under
+* the terms of the Eclipse Public License 2.0 which accompanies this
+* distribution and is available at http://eclipse.org/legal/epl-2.0
+* or the Apache License, Version 2.0 which accompanies this distribution
+* and is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the following Secondary
+* Licenses when the conditions for such availability set forth in the
+* Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+* version 2 with the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*******************************************************************************/
+
+#include "optimizer/RecognizedCallTransformer.hpp"
+
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+#include "il/TreeTop.hpp"
+#include "il/TreeTop_inlines.hpp"
+#include "codegen/CodeGenerator.hpp"                      // for CodeGenerator
+#include "il/ILOpCodes.hpp"
+#include "il/ILOps.hpp"                                   // for ILOpCode, etc
+#include "ilgen/IlGenRequest.hpp"

```

Comment:
Why are these needed - it is unusual for an optimization to have dependencies on the ilgenerator...

Reply:
Updated.